src/t8_geometry/t8_geometry_helpers.c

```diff
@@ -372,6 +372,223 @@ t8_geom_get_triangle_scaling_factor (int edge_index, const double *tree_vertices
+  case 3:
+    switch (face) {
+    case 0:
+      orthogonal_direction = 1 - ref_coords[2];
+      max_orthogonal_direction = 1;
+      break;
+    case 4:
+      orthogonal_direction = 1 - ref_coords[0];
+      max_orthogonal_direction = 1 - ref_coords[1];
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+      break;
+    }
+    break;
+  case 4:
+    switch (face) {
+    case 1:
+      orthogonal_direction = 1 - ref_coords[2];
+      max_orthogonal_direction = 1;
+      break;
+    case 4:
+      orthogonal_direction = ref_coords[0] - ref_coords[1];
+      max_orthogonal_direction = ref_coords[0];
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+      break;
+    }
+    break;
+  case 5:
+    switch (face) {
+    case 2:
+      orthogonal_direction = 1 - ref_coords[2];
+      max_orthogonal_direction = 1;
+      break;
+    case 4:
+      orthogonal_direction = ref_coords[1];
+      max_orthogonal_direction = ref_coords[0];
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+      break;
+    }
+    break;
+  case 6:
+    switch (face) {
+    case 0:
+      orthogonal_direction = ref_coords[1];
+      max_orthogonal_direction = 1;
+      break;
+    case 2:
+      orthogonal_direction = 1 - ref_coords[0];
+      max_orthogonal_direction = 1;
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+      break;
+    }
+    break;
+  case 7:
+    switch (face) {
+    case 0:
+      orthogonal_direction = 1 - ref_coords[1];
+      max_orthogonal_direction = 1;
+      break;
+    case 1:
+      orthogonal_direction = 1 - ref_coords[0];
+      max_orthogonal_direction = 1;
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+      break;
+    }
+    break;
+  case 8:
+    switch (face) {
+    case 1:
+      orthogonal_direction = ref_coords[0];
+      max_orthogonal_direction = 1;
+      break;
+    case 2:
+      orthogonal_direction = ref_coords[0];
+      max_orthogonal_direction = 1;
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+      break;
+    }
+    break;
+  default:
+    SC_ABORT_NOT_REACHED ();
+    break;
+  }
+  /* If the maximum orthogonal direction is 1, the reference coordinate lies on
+   * one of the edge nodes and the scaling factor is therefore 0, because the displacement
+   * at the nodes is always 0.
+   * In all other cases the scaling factor is determined with one minus the relation of the orthogonal direction
+   * to the maximum orthogonal direction. */
+  if (max_orthogonal_direction == 0) {
+    scaling_factor = 0;
+  }
+  else {
+    scaling_factor = 1 - (orthogonal_direction / max_orthogonal_direction);
+  }
+  return scaling_factor;
+}
+
+double
+t8_geom_get_scaling_factor_face_through_volume_prism (const int face, const double *ref_coords)
+{
+  double scaling_factor;
+
+  switch (face) {
+  case 0:
+    if (ref_coords[1] == 1) {
+      scaling_factor = 0;
+    }
+    else {
+      scaling_factor = 1 - ((1 - ref_coords[0]) / (1 - ref_coords[1]));
```

Comment:
```suggestion
      scaling_factor = (ref_coords[0] - ref_coords[1]) / (1 - ref_coords[1]);
```
please double check my reformulation.